### PR TITLE
feat(domain): add encryption for merchant account

### PR DESCRIPTION
**Pull request by dracarys18**
## Type of Change

- [x] Enhancement

## Description

Adding encryption fields for merchant account.

## How did you test it?

Manual

## Checklist

- [x] I formatted the code `cargo +nightly fmt --all`
- [x] I addressed lints thrown by `cargo clippy`
- [x] I reviewed submitted code

### CODE REVIEW

File: crates/api_models/src/admin.rs [hunk 1]
```diff
@@ -132,12 +132,12 @@ pub struct MerchantAccountResponse {
     pub merchant_id: String,
 
     /// Name of the Merchant Account
-    #[schema(example = "NewAge Retailer")]
-    pub merchant_name: Option<String>,
+    #[schema(value_type = Option<String>,example = "NewAge Retailer")]
+    pub merchant_name: Option<Encryptable<Secret<String>>>,
```
Comment (NishantJoshi00) [inline crates/api_models/src/admin.rs:136]: Use the `OptionalEncryptableName`

File: crates/api_models/src/admin.rs [hunk 2]
```diff
@@ -132,12 +132,12 @@ pub struct MerchantAccountResponse {
...
 
     /// API key that will be used for server side API access
     #[schema(value_type = Option<String>, example = "Ah2354543543523")]
-    pub api_key: Option<StrongSecret<String>>,
+    pub api_key: Option<Encryptable<Secret<String>>>,
```
Comment (NishantJoshi00) [inline crates/api_models/src/admin.rs:140]: Same here

File: crates/router/src/types/domain/merchant_account.rs [hunk 3]
```diff
@@ -1,19 +1,29 @@
...
 pub struct MerchantAccount {
     pub id: Option<i32>,
     pub merchant_id: String,
     pub return_url: Option<String>,
     pub enable_payment_response_hash: bool,
     pub payment_response_hash_key: Option<String>,
     pub redirect_to_merchant_with_http_post: bool,
-    pub merchant_name: Option<String>,
-    pub merchant_details: Option<serde_json::Value>,
+    pub merchant_name: Option<Encryptable<Secret<String>>>,
```
Comment (NishantJoshi00) [inline crates/router/src/types/domain/merchant_account.rs:25]: same here

**Review by NishantJoshi00 — CHANGES_REQUESTED** [review]

**Review by NishantJoshi00 — APPROVED** [review]

File: crates/router/src/core/admin.rs [hunk 4]
```diff
@@ -63,11 +75,12 @@ pub async fn create_merchant_account(
             .attach_printable("Unexpected create API key response"),
     }?;
 
-    let merchant_details = Some(
+    let merchant_details: Option<masking::Secret<serde_json::Value>> = Some(
```
Comment (jarnura) [inline crates/router/src/core/admin.rs:78]: use type alias for this Option<masking::Secret<serde_json::Value>>

File: crates/router/src/core/admin.rs [hunk 5]
```diff
@@ -170,13 +199,22 @@ pub async fn merchant_account_update(
     }
 
     let updated_merchant_account = storage::MerchantAccountUpdate::Update {
-        merchant_name: req.merchant_name,
+        merchant_name: req
+            .merchant_name
+            .async_map(|inner| crypto::Encryptable::encrypt(inner.into(), &key, GcmAes256 {}))
```
Comment (jarnura) [inline crates/router/src/core/admin.rs:204]: use async_lift instead if async_map.

Comment (jarnura) [reply crates/router/src/core/admin.rs:204]: Update in other places too

**Review by jarnura — CHANGES_REQUESTED** [review]

**Review by jarnura — APPROVED** [review]

**Pull request merged**